File: app/src/main/java/com/gmail/fattazzo/formula1world/view/chart/marker/F1MarkerLapTimeView.java
package com.gmail.fattazzo.formula1world.view.chart.marker;

import android.content.Context;
import android.widget.TextView;

import com.github.mikephil.charting.components.MarkerView;
import com.github.mikephil.charting.data.Entry;
import com.github.mikephil.charting.highlight.Highlight;
import com.github.mikephil.charting.utils.MPPointF;
import com.gmail.fattazzo.formula1world.R;
import com.gmail.fattazzo.formula1world.domain.F1LapTime;

public class F1MarkerLapTimeView extends MarkerView {

    private TextView tvContent;
    private TextView tvContent2;

    public F1MarkerLapTimeView(Context context) {
        super(context, R.layout.chart_marker_view_large);

        tvContent = (TextView) findViewById(R.id.tvContent);
        tvContent2 = (TextView) findViewById(R.id.tvContent2);
    }

    @Override
    public void refreshContent(Entry e, Highlight highlight) {

        if (e.getData() instanceof F1LapTime) {
            F1LapTime lapTime = (F1LapTime) e.getData();

            tvContent.setText(lapTime.driver.getFullName());
            tvContent2.setText(lapTime.time);
        } else {
            tvContent.setText("");
            tvContent2.setText("");
        }

        super.refreshContent(e, highlight);
    }

    @Override
    public MPPointF getOffset() {
        return new MPPointF(-(getWidth() / 2), -getHeight());
    }
}

File: app/src/main/kotlin/com/gmail/fattazzo/formula1world/view/chart/marker/F1MarkerLapTimeView.kt
package com.gmail.fattazzo.formula1world.view.chart.marker

import android.content.Context
import android.widget.TextView

import com.github.mikephil.charting.components.MarkerView
import com.github.mikephil.charting.data.Entry
import com.github.mikephil.charting.highlight.Highlight
import com.github.mikephil.charting.utils.MPPointF
import com.gmail.fattazzo.formula1world.R
import com.gmail.fattazzo.formula1world.domain.F1LapTime

class F1MarkerLapTimeView(context: Context) : MarkerView(context, R.layout.chart_marker_view_large) {

    private val tvContent: TextView
    private val tvContent2: TextView

    init {

        tvContent = findViewById(R.id.tvContent) as TextView
        tvContent2 = findViewById(R.id.tvContent2) as TextView
    }

    override fun refreshContent(e: Entry?, highlight: Highlight?) {

        if (e!!.data is F1LapTime) {
            val lapTime = e.data as F1LapTime

            tvContent.text = lapTime.driver!!.fullName
            tvContent2.text = lapTime.time
        } else {
            tvContent.text = ""
            tvContent2.text = ""
        }

        super.refreshContent(e, highlight)
    }

    override fun getOffset(): MPPointF {
        return MPPointF((-(width / 2)).toFloat(), (-height).toFloat())
    }
}
